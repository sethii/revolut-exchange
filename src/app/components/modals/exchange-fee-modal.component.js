import React from 'react';
import PropTypes from 'prop-types';
import { Modal, Header, Link } from './modal.component';

import { branch, renderNothing, setPropTypes, compose } from 'recompose';

const enhance = compose(
  setPropTypes({
    display: PropTypes.bool.isRequired,
    closeModal: PropTypes.func.isRequired
  }),
  branch(
    ({ display }) => !display,
    renderNothing
  )
);

export const ExchangeFeeModal = enhance(({ closeModal }) => <Modal>
  <Header>FX exchange limit fee</Header>
  <p>You've reached FX rate limit of $6000 per month. Future exchange will proceed with 0,5% fee.</p>

  <Link onClick={closeModal}>GOT IT</Link>
</Modal>);
