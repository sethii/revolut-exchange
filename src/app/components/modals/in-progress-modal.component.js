import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { exchangeCurrencyFinished } from '../../redux';
import { Modal, Header, Button } from './modal.component';
import { branch, renderNothing, setPropTypes, compose } from 'recompose';

const enhance = compose(
  connect(
    null,
    (dispatch) => ({
      done: () => dispatch(exchangeCurrencyFinished())
    })
  ),
  setPropTypes({
    display: PropTypes.bool.isRequired,
    done: PropTypes.func.isRequired
  }),
  branch(
    ({ display }) => !display,
    renderNothing
  )
);

export const InProgressModal = enhance(({ done }) => {
  return (
    <Modal>
      <Header>Exchange finished</Header>
      <Button onClick={() => { done(); }}>Close</Button>
    </Modal>
  )
});

