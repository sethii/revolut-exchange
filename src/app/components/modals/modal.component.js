import styled from 'react-emotion';

export const Modal = styled.div`
  border-radius: 10px;
  width: 500px;
  height: auto;
  padding: 20px;
  background-color: white;
  margin: 0 auto;
  font-size: 14px;
`;

export const Header = styled.h1`
  font-size: 24px;
`;

export const Link = styled.a`
  font-size: 16px;
  color: #4169E1;
  
  &:hover {
    cursor: pointer;
  }
`;

export const Button = styled.button`
  border: none;
  font-size: 16px;
  background-color: #4169E1;
  color: white;
  padding: 15px 10px;
  width: 100%;
  border-radius: 25px;
  
  &:hover {
    cursor: pointer;
  }
`;
