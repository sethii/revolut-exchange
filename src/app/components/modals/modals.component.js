import React from 'react';

import { closeModal } from '../../redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { branch, compose, renderNothing, setPropTypes, defaultProps } from 'recompose';

import { ExchangeFeeModal } from './exchange-fee-modal.component';
import { InProgressModal } from './in-progress-modal.component';
import styled from 'react-emotion';

const enhance = compose(
  connect(
    (state) => ({
      display: typeof state.app.modal.type !== 'undefined',
      type: state.app.modal.type
    }),
    (dispatch) => ({
      closeModal: () => dispatch(closeModal())
    })
  ),
  setPropTypes({
    display: PropTypes.bool.isRequired,
    type: PropTypes.string,
    closeModal: PropTypes.func.isRequired
  }),
  defaultProps({
    type: ''
  }),
  branch(
    ({ display }) => !display,
    renderNothing
  )
);

const ModalsBackground = styled.div`
  display: flex;
  justify-content: center;
  align-items:center;
  position: fixed;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.9);
`;

export const Modals = enhance(({ type, closeModal }) => <ModalsBackground>
  <ExchangeFeeModal display={type === 'exchange-fee-modal'} closeModal={closeModal}/>
  <InProgressModal display={type === 'in-progress-modal'}/>
</ModalsBackground>);
