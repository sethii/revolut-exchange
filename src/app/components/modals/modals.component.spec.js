import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { toClass } from 'recompose';

import { Modals } from './modals.component';

describe('Modals component', () => {
  it('renders in modals component', () => {
    const ModalsComponent = toClass(Modals);
    const fakeStore = {
      getState: () => ({
        app: {
          modal: {
            type: 'some-modal'
          }
        }
      }),
      dispatch: () => jest.fn(),
      subscribe: () => jest.fn()
    };

    expect(renderer.create(<Provider store={fakeStore}><ModalsComponent/></Provider>).toJSON()).toMatchSnapshot();
  });

  it('renders null when display false passed', () => {
    const ModalsComponent = toClass(Modals);
    const fakeStore = {
      getState: () => ({
        app: {
          modal: {}
        }
      }),
      dispatch: () => jest.fn(),
      subscribe: () => jest.fn()
    };

    expect(renderer.create(<Provider store={fakeStore}><ModalsComponent/></Provider>).toJSON()).toMatchSnapshot();
  });

  it('renders in progress modal modal when in-progress-modal type given', () => {
    const ModalsComponent = toClass(Modals);
    const fakeStore = {
      getState: () => ({
        app: {
          modal: {
            type: 'in-progress-modal'
          }
        }
      }),
      dispatch: () => jest.fn(),
      subscribe: () => jest.fn()
    };

    expect(renderer.create(<Provider store={fakeStore}><ModalsComponent/></Provider>).toJSON()).toMatchSnapshot();
  });

  it('renders exchange fee modal modal when exchange-fee-modal type given', () => {
    const ModalsComponent = toClass(Modals);
    const fakeStore = {
      getState: () => ({
        app: {
          modal: {
            type: 'exchange-fee-modal'
          }
        }
      }),
      dispatch: () => jest.fn(),
      subscribe: () => jest.fn()
    };

    expect(renderer.create(<Provider store={fakeStore}><ModalsComponent/></Provider>).toJSON()).toMatchSnapshot();
  });
});
