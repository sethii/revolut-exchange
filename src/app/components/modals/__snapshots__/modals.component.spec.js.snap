// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modals component renders exchange fee modal modal when exchange-fee-modal type given 1`] = `
.glamor-7 {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: fixed;
  top: 0px;
  bottom: 0px;
  right: 0px;
  left: 0px;
  background-color: rgba(0, 0, 0, 0.9);
}

.glamor-5 {
  border-radius: 10px;
  width: 500px;
  height: auto;
  padding: 20px;
  background-color: white;
  margin: 0 auto;
  font-size: 14px;
}

.glamor-1 {
  font-size: 24px;
}

.glamor-3 {
  font-size: 16px;
  color: #4169E1;
}

.glamor-3:hover {
  cursor: pointer;
}

<div
  className="glamor-6 glamor-7"
>
  <div
    className="glamor-4 glamor-5"
  >
    <h1
      className="glamor-0 glamor-1"
    >
      FX exchange limit fee
    </h1>
    <p>
      You've reached FX rate limit of $6000 per month. Future exchange will proceed with 0,5% fee.
    </p>
    <a
      className="glamor-2 glamor-3"
      onClick={[Function]}
    >
      GOT IT
    </a>
  </div>
</div>
`;

exports[`Modals component renders in modals component 1`] = `
.glamor-1 {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: fixed;
  top: 0px;
  bottom: 0px;
  right: 0px;
  left: 0px;
  background-color: rgba(0, 0, 0, 0.9);
}

<div
  className="glamor-0 glamor-1"
/>
`;

exports[`Modals component renders in progress modal modal when in-progress-modal type given 1`] = `
.glamor-7 {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: fixed;
  top: 0px;
  bottom: 0px;
  right: 0px;
  left: 0px;
  background-color: rgba(0, 0, 0, 0.9);
}

.glamor-5 {
  border-radius: 10px;
  width: 500px;
  height: auto;
  padding: 20px;
  background-color: white;
  margin: 0 auto;
  font-size: 14px;
}

.glamor-1 {
  font-size: 24px;
}

.glamor-3 {
  border: none;
  font-size: 16px;
  background-color: #4169E1;
  color: white;
  padding: 15px 10px;
  width: 100%;
  border-radius: 25px;
}

.glamor-3:hover {
  cursor: pointer;
}

<div
  className="glamor-6 glamor-7"
>
  <div
    className="glamor-4 glamor-5"
  >
    <h1
      className="glamor-0 glamor-1"
    >
      Exchange finished
    </h1>
    <button
      className="glamor-2 glamor-3"
      onClick={[Function]}
    >
      Close
    </button>
  </div>
</div>
`;

exports[`Modals component renders null when display false passed 1`] = `null`;
